### PR TITLE
PF-2258: fix flaky test ControlledGcpResourceApiControllerGcsBucketTest

### PR DESCRIPTION
**Pull request by yuhuyoyo**
ControlledGcpResourceApiControllerGcsBucketTest#clone_copyNothing

When I'm running this test locally, the cloning finishes so fast that it returns 200 consistently for me.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 1]
```diff
@@ -910,14 +909,44 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucket(
...
-  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsync(
+  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketWithError(
       AuthenticatedUserRequest userRequest,
       UUID sourceWorkspaceId,
       UUID sourceResourceId,
       UUID destWorkspaceId,
       ApiCloningInstructionsEnum cloningInstructions,
       String destBucketName,
-      int expectedCode)
+      int statusCode)
```
Comment (melissachang) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:919]: Can this be expectedCode, to be a little more clear?

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 2]
```diff
@@ -938,14 +967,12 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsync(
                                 sourceWorkspaceId, sourceResourceId))
                             .content(objectMapper.writeValueAsString(request)),
                         userRequest)))
-            .andExpect(status().is(expectedCode))
+            .andExpect(status().is2xxSuccessful())
             .andReturn()
             .getResponse()
             .getContentAsString();
     // If an exception was thrown, deserialization won't work, so don't attempt it.
```
Comment (melissachang) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:974]: Delete

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 3]
```diff
@@ -910,14 +909,44 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucket(
   }
 
   /** Call cloneGcsBucket() and return immediately; don't wait for flight to finish. */
-  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsync(
+  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketWithError(
```
Comment (melissachang) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:912]: Can we use these names, to be a little more consistent and succinct?

cloneControlledGcsBucketAsyncError

cloneControlledGcsBucketAsyncSuccess

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 4]
```diff
@@ -910,14 +909,44 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucket(
...
+                    post(CLONE_CONTROLLED_GCP_GCS_BUCKET_FORMAT.formatted(
+                            sourceWorkspaceId, sourceResourceId))
+                        .content(objectMapper.writeValueAsString(request)),
+                    userRequest)))
+        .andExpect(status().is(statusCode));
+    // If an exception was thrown, deserialization won't work, so don't attempt it.
+    return null;
+  }
+
+  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsyncSuccessfully(
```
Comment (melissachang) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:943]: // This checks 200 and 204. (For some tests, sometimes the test will return 200 and other times 204.)

File: service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java [hunk 5]
```diff
@@ -910,14 +909,44 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucket(
...
+    mockMvc
+        .perform(
+            addJsonContentType(
+                addAuth(
+                    post(CLONE_CONTROLLED_GCP_GCS_BUCKET_FORMAT.formatted(
+                            sourceWorkspaceId, sourceResourceId))
+                        .content(objectMapper.writeValueAsString(request)),
+                    userRequest)))
+        .andExpect(status().is(statusCode));
+    // If an exception was thrown, deserialization won't work, so don't attempt it.
```
Comment (melissachang) [inline service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java:939]: Delete 939-940. Change method to return void.

**Review by melissachang — APPROVED** [review]
Thanks!

**Review by ddietterich — APPROVED** [review]

**Pull request merged**